### PR TITLE
test(inference): add finite-difference gradient validation suite

### PR DESCRIPTION
**Pull request by TobiBu**
## Summary
- add reusable finite-difference gradient utility for pytrees
- add gradient comparison metrics helper
- add tests comparing autodiff and finite-difference gradients
- cover deterministic and fixed-key stochastic settings

## Stack position
PR 5 / 8 in the gradient-production plan

### CODE REVIEW

File: rubix/inference/validation.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        Any: Pytree gradient matching the structure of ``params``.
+    """
+    if eps <= 0:
+        raise ValueError("eps must be strictly positive")
+
+    flat, unravel = ravel_pytree(params)
+
+    def f_flat(flat_params):
+        return loss_fn(unravel(flat_params))
+
```
[the remark this answers is not in the document]
Comment (TobiBu) [reply rubix/inference/validation.py:55]: @copilot apply changes based on [this feedback](https://github.com/AstroAI-Lab/rubix/pull/202#discussion_r3100939395)

File: tests/test_inference_gradient_validation.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+            "metallicity": jnp.array([3.0]),
+        }
+    }
+
+    comparison = compare_gradients(gradient, gradient)
+
+    assert comparison.max_abs_error == 0.0
+    assert comparison.l2_error == 0.0
+    assert comparison.relative_l2_error == 0.0
+
```
[the remark this answers is not in the document]
Comment (TobiBu) [reply tests/test_inference_gradient_validation.py:138]: @copilot apply changes based on [this feedback](https://github.com/AstroAI-Lab/rubix/pull/202#discussion_r3100939475)

**Comment by TobiBu** [comment]
@copilot please fix the doctoring according to this [pre-commit failure](https://results.pre-commit.ci/run/github/781992985/1776679343.-hUaUbrrRF6YtIObdQpQpQ)
